evm/bridge-token-factory/contracts/BridgeTokenFactoryWormhole.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import {BridgeTokenFactory} from "./BridgeTokenFactory.sol";
+import "./Borsh.sol";
+
+interface IWormhole {
+    function publishMessage(
+        uint32 nonce,
+        bytes memory payload,
+        uint8 consistencyLevel
+    ) external payable returns (uint64 sequence);
+}
+
+contract BridgeTokenFactoryWormhole is BridgeTokenFactory {
+    IWormhole private _wormhole;
+    uint8 private _consistencyLevel;
+    uint32 public wormholeNonce;
+
+    function initializeWormhole(
+        address _tokenImplementationAddress,
+        address _nearBridgeDerivedAddress,
+        address wormholeAddress,
+        uint8 consistencyLevel
+    ) external initializer {
+        initialize(_tokenImplementationAddress, _nearBridgeDerivedAddress);
+        _wormhole = IWormhole(wormholeAddress);
+        _consistencyLevel = consistencyLevel;
+    }
+
+    function depositExtension(BridgeDeposit memory bridgeDeposit) internal override {
+        _wormhole.publishMessage(
+            wormholeNonce,
+            abi.encode(bridgeDeposit.token, bridgeDeposit.amount, bridgeDeposit.feeRecipient, bridgeDeposit.nonce),
```

Comment:
Maybe use `encodeWithSignature` for encoding and specify what kind of event it is, so that it's easier to decode using the prefix.

Reply:
https://github.com/Near-One/omni-bridge/blob/main/near/omni-types/src/prover_result.rs#L15

Reply:
Not a bad idea, @karim-en ?
It can be useful for example if we read all events from Wormhole API so we don't know the type of the events

Reply:
yeh, let's add uint8 at the beginning

Reply:
Added ee552d5